src/main/java/spoon/support/reflect/declaration/CtRecordImpl.java

```diff
@@ -92,6 +100,51 @@ && getField(component.getSimpleName()).isImplicit()) {
 		return this;
 	}
 
+	@Override
+	public CtRecord createCanonicalConstructorIfMissing() {
+
+		CtTypeReference<?>[] typeReferences =
+			getFields().stream()
+				.filter(CtElement::isImplicit)
+				.map(CtTypedElement::getType)
+				.toList().toArray(new CtTypeReference[0]);
+
+		// If a canonical constructor is not already present, adds one
+		CtConstructor<?> constructor = getConstructor(typeReferences);
+		if (constructor == null) {
+			CtConstructor<?> canonical = getFactory().createConstructor();
+			canonical.setImplicit(true);
+			canonical.setExtendedModifiers(Set.of(CtExtendedModifier.implicit(ModifierKind.PUBLIC)));
```

Comment:
According to <https://docs.oracle.com/javase/specs/jls/se25/html/jls-8.html#jls-8.10.4>, the modifier should correspond to the visibility of the record itself. I guess the easiest would be to use `ModifierTarget.CONSTRUCTOR.getAllowedKinds()`, check if the modifier is existent on the record, and apply it on the constructor if so.